src/wallets/IdentityWallet.ts

```diff
@@ -480,3 +484,22 @@ export function calculateIdentityAddress(
       .slice(2 + 2 * 12)
   return toChecksumAddress(address)
 }
+
+/**
+ * Generates a random nonce to use for meta transactions.
+ * The nonce fits into the range of ]2^255, 2^256[.
+ * This is an alternative to the up counting nonce (]0, 2^255[) without the need
+ * to query a [[TLProvider]].
+ */
+export function getRandomNonce(): string {
+  const random = BigNumber.random()
+  const nonce = random.multipliedBy(RANDOM_NONCE_RANGE).plus(MIN_RANDOM_NONCE)
```

Comment:
You should add 1 to this, I am not sure what you do with the reduce later, maybe you do it there.
The random number generated can be `0`, in which case the nonce is `MIN_RANDOM_NONCE` which is too low.
However if you add `1`, the nonce generated can be `2**255 - 1 + 2*255 + 1 = 2**256` which is too high, so you should lower the range by `1`.

I would suggest using `MIN_RANDOM_NONCE = new BigNumber(2).pow(255).plus(1)`

Reply:
Yes, this is correct. I obviously missed that even I knew.